pkg/healthcheck/healthcheck.go

```diff
@@ -1257,7 +1393,13 @@ func (hc *HealthChecker) expectedRBACNames() []string {
 	}
 }
 
-func (hc *HealthChecker) checkClusterRoles(shouldExist bool) error {
+func (hc *HealthChecker) expectedCniRBACNames() []string {
+	return []string{
+		"linkerd-cni",
+	}
+}
+
+func (hc *HealthChecker) checkClusterRoles(shouldExist bool, expectedRoles []string, skipCni bool) error {
```

Comment:
I agree the omission of the parameter. however I think having a seprate section of CNI checks that contains a check for the cluster role specific to CNI makes more sense rather than combining it into the control plane checks. Wouldn't you say so?

Reply:
In fact looking at this again, here is why I added this bool parameter and I think it is needed. So there are three cases where these methods are hit

- LinkerdPreInstallGlobalResourcesChecks - this is run during install. This needs to skip the CNI resources if cni is enabled (depends on hc.NoInitContainer)
- LinkerdConfigChecks - same as above really
- LinkerdCniChecks - here hc.NoInitContainer is true which means cni is enabled. Nevertheless we should not be skipping these resources because thats what we are checking for. 

So we cannot purely rely on the values of NoInitContainers to determine what we need to do because its interpretation changes based on the check that we are running. Therefore the extra bool.  The whole thing kind of stinks to me, I am not sure how to make it simpler so if you have any suggestions feel free to share.